pkg/trace/config/config.go

```diff
@@ -287,10 +287,9 @@ type AgentConfig struct {
 	// Concentrator
 	BucketInterval         time.Duration // the size of our pre-aggregation per bucket
 	ExtraAggregators       []string      // DEPRECATED
-	PeerServiceAggregation bool          // TO BE DEPRECATED - enables/disables stats aggregation for peer.service, used by Concentrator and ClientStatsAggregator
```

Comment:
cc @mx-psi for context (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34177)

We can leave this config in datadog exporter and map it to `PeerTagsAggregation` for now, but eventually it should go away.

Reply:
Is the option still going to be available in the Agent?

Reply:
Alright, I think we can align the removal in both projects then